src/common/decorators/swagger.ts

```diff
@@ -115,22 +115,32 @@ export function ApiTransactionResponse(
   );
 }
 
+type TransactionResponseMessages = Record<
+  'notFound' | 'badRequest' | 'unprocessableEntity',
+  string
+>;
+
 /**
  * A helper that combines responses for SDK Error like `BadRequestException`, `NotFoundException`, `UnprocessableEntityException`, `InternalServerErrorException`
  *
  * @param options - these will be passed to the `ApiNotFoundResponse` decorator
  */
 export function ApiTransactionFailedResponse(
-  options: ApiResponseOptions
+  messages: Partial<TransactionResponseMessages>
 ): ReturnType<typeof applyDecorators> {
+  const {
+    notFound = 'Returned if the Entity was not found',
```

Comment:
Whats the intention here?

I think its better to only have messages for what is given, not default them to some value. Not every endpoint returns every error, so we should conditionally apply the correct decorators. As in pass `applyDecorators` a variable.

I think if `TransactionResponseMessages = Record<HttpStatus, string | undefined>` then the controller can pass in the status but not supply a message for the default

Reply:
FWIW I do like this decorator and think it can clean up the controllers, just that I don't want to say an endpoint returns "Unprocessable Entity" when it never does.